File: IsometricShooter/Scripts/CS/Material/MaterialByAngleY.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

public class MaterialByAngleY : MonoBehaviour {

	public Transform directionReference;
	public MeshRenderer materialTarget;
	public List<DirectionMaterial> materialList; //list of DirectionMaterial

	private Material currentMaterial;
	
	// Update is called once per frame
	void Update () {
	
		for(int i = 0; i < materialList.Count; i++) {
			DirectionMaterial m = materialList[i];
			if(directionReference.eulerAngles.y >= m.fromAngle
			&& directionReference.eulerAngles.y <= m.toAngle) {
				if(currentMaterial != m.material) {
					currentMaterial = m.material;
					materialTarget.material = currentMaterial;
				}
			}
		}
	
	}
}

File: IsometricShooter/Scripts/JS/Material/MaterialByAngleY.js
#pragma strict

import System.Collections.Generic; //enable lists

var directionReference : Transform;
var materialTarget : MeshRenderer;
var materialList : List.<DirectionMaterial>; //list of DirectionMaterial

private var currentMaterial : Material;

function Update () {

	for(var i : int = 0; i < materialList.Count; i++) {
		var m = materialList[i];
		if(directionReference.eulerAngles.y >= m.fromAngle
		&& directionReference.eulerAngles.y <= m.toAngle) {
			if(currentMaterial != m.material) {
				currentMaterial = m.material;
				materialTarget.material = currentMaterial;
			}
		}
	}

}